src/serialbox-c/FortranWrapper.cpp

```diff
@@ -154,35 +170,33 @@ void serialboxFortranSerializerGetFieldDimensions(const void* serializer, const
   const Serializer* ser = toConstSerializer(static_cast<const serialboxSerializer_t*>(serializer));
 
   try {
-    const auto& dims = ser->getFieldMetainfoImplOf(name).dims();
+    auto dims = ser->getFieldMetainfoImplOf(name).dims();
 
-    if(dims.size() != 4)
-      throw Exception("number of dimensions is %i, required are 4", dims.size());
+    ::make_4D(dims);
 
-	*isize = dims[0];
-	*jsize = dims[1];
-	*ksize = dims[2];
-	*lsize = dims[3];
-  }
-  catch(std::exception& e) {
+    *isize = dims[0];
+    *jsize = dims[1];
+    *ksize = dims[2];
+    *lsize = dims[3];
+  } catch(std::exception& e) {
```

Comment:
The only possible source of an exception here is allocating new memory in the vector due to `push_back` or `resize`. This will give an hardly understandable error message. In the previous version this try/catch block was used for a wrong number of dimensions passed, which would give an error message directly related to a bad usage of the API.

I would suggest to either:
- create a new, clearer exception with a message saying that an internal error occurred, or
- just remove the try/catch block; in the improbable case that an allocation for 4 values will fail, it will trigger a crash which the user can with low effort relate to an internal problem in Serialbox.

But it's not a strong opinion. I'm just being picky here...

Reply:
I just missed that. I will remove the `try/catch`.